### PR TITLE
Redesign of the "Install Package" page

### PR DESCRIPTION
**Pull request by dtdesign**
The search feature was pretty much unchanged in the previous versions and dates back to the early stages of WCF 2. While it did work somewhat fine, its UX was sometimes unpredictable and provided the user with little visual feedback.

The new UI is a bit more simplified by combining the different search fields (did you knew it was an `AND` search?) into a single input. Uploading packages is now handled through a button at the top, putting more emphasis on the package search, which does an amazing job at figuring out which packages can actually be installed.

### CODE REVIEW

File: wcfsetup/install/files/acp/templates/packageSearchResultList.tpl [hunk 1]
```diff
@@ -1,43 +1,53 @@
...
-				
-				{event name='headColumns'}
+				<td colspan="5"><small>{lang count=$thirdPartySources|count}wcf.acp.package.search.result.thirdParty{/lang}</small></td>
 			</tr>
-		</thead>
-		
-		<tbody>
 			{content}
-				{foreach from=$packageUpdates item=$package}
+				{foreach from=$thirdPartySources item=$package}
```
Comment (joshuaruesweg) [inline wcfsetup/install/files/acp/templates/packageSearchResultList.tpl]: When searching for a package, the following error is thrown (`PackageUpdateAction::search` does not assign the variable to the template, if no package are found):  
```
Undefined index: trustedSources


Stacktrace:

#0 /Applications/MAMP/htdocs/hurricane/acp/templates/compiled/0_wcf_1_packageSearchResultList.php(30): wcf\system\WCF::handleError(8, 'Undefined index...', '/Applications/M...', 30, Array)
#1 /Applications/MAMP/htdocs/hurricane/lib/system/template/TemplateEngine.class.php(321): include('/Applications/M...')
#2 /Applications/MAMP/htdocs/hurricane/lib/system/template/TemplateEngine.class.php(549): wcf\system\template\TemplateEngine->display('packageSearchRe...', 'wcf', false)
#3 /Applications/MAMP/htdocs/hurricane/lib/data/package/update/PackageUpdateAction.class.php(99): wcf\system\template\TemplateEngine->fetch('packageSearchRe...')
#4 [internal function]: wcf\data\package\update\PackageUpdateAction->search()
#5 /Applications/MAMP/htdocs/hurricane/lib/data/AbstractDatabaseObjectAction.class.php(204): call_user_func(Array)
#6 /Applications/MAMP/htdocs/hurricane/lib/action/AJAXProxyAction.class.php(75): wcf\data\AbstractDatabaseObjectAction->executeAction()
#7 /Applications/MAMP/htdocs/hurricane/lib/action/AJAXInvokeAction.class.php(94): wcf\action\AJAXProxyAction->invoke()
#8 /Applications/MAMP/htdocs/hurricane/lib/action/AbstractAction.class.php(47): wcf\action\AJAXInvokeAction->execute()
#9 /Applications/MAMP/htdocs/hurricane/lib/action/AJAXInvokeAction.class.php(61): wcf\action\AbstractAction->__run()
#10 /Applications/MAMP/htdocs/hurricane/lib/system/request/Request.class.php(83): wcf\action\AJAXInvokeAction->__run()
#11 /Applications/MAMP/htdocs/hurricane/lib/system/request/RequestHandler.class.php(96): wcf\system\request\Request->execute()
#12 /Applications/MAMP/htdocs/hurricane/acp/index.php(9): wcf\system\request\RequestHandler->handle('wcf', true)
#13 {main}```

File: wcfsetup/install/files/js/WoltLabSuite/Core/Acp/Ui/Package/PrepareInstallation.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+						saveCredentials: elById('packageUpdateServerSaveCredentials').checked,
+						username: username
+					});
+				}
+			}
+		},
+		
+		_ajaxSuccess: function(data) {
+			if (data.returnValues.queueID) {
+				UiDialog.close(this);
```
Comment (joshuaruesweg) [inline wcfsetup/install/files/js/WoltLabSuite/Core/Acp/Ui/Package/PrepareInstallation.js]: When I trie to install a package, I get the following error: 
```
Uncaught Error: Expected a valid dialog id, '[object Object]' does not match any active dialog.
    at Object.close (Dialog.js?t=1550865154:792)
    at AcpUiPackagePrepareInstallation._ajaxSuccess (PrepareInstallation.js?t=1550865154:76)
    at AjaxRequest._success (Request.js?t=1550865154:257)
    at XMLHttpRequest._xhr.onload (Request.js?t=1550865154:135)
```

Comment (dtdesign) [reply wcfsetup/install/files/js/WoltLabSuite/Core/Acp/Ui/Package/PrepareInstallation.js]: Thanks, fixed.

File: wcfsetup/install/files/acp/templates/packageSearchResultList.tpl (hunk 1 above)
Comment (dtdesign) [reply wcfsetup/install/files/acp/templates/packageSearchResultList.tpl]: Thanks, fixed.

**Comment by dtdesign** [comment]
The description has always been available as a tooltip. I did include it in an earlier draft, but I was anything but happy with them, they're pretty useless on average and thus increase the visual clutter.

**Pull request merged**